app/src/org/commcare/utils/FileUtil.java

```diff
@@ -553,22 +526,55 @@ public static String getExtension(String filePath) {
     /**
      * Retrieve a file's name using contentUri.
      * https://developer.android.com/training/secure-file-sharing/retrieve-info.html#RetrieveFileInfo
+     *
+     * @throws FileExtensionNotFoundException
      */
     public static String getFileName(Context context, Uri uri) {
+        String fileName;
         try (Cursor cursor = context.getContentResolver().query(uri, null, null, null, null)) {

```

Comment:
This should have a projection like the above which limits the data retrieved to `OpenableColumns.DISPLAY_NAME` to reduce the amount of unnecessary data read from the provider

Reply:
@ShivamPokhriyal this looks unaddressed.

Reply:
Oops, seems like I missed it.